File: DemoServer/AdditionalLanguages/go/Advanced/createDatabase/CreateDatabase.go
package main

//region Usings
import {
    "fmt" 
    "github.com/ravendb/ravendb-go-client"
}
//endregion

func main() {
    createDatabase()
}

//region Demo
func createDatabase() error {

    //region Step_1
    databaseRecord := ravendb.NewDatabaseRecord()
    databaseRecord.DatabaseName = "NameOfDatabase"
    
    createDatabaseOperation := ravendb.NewCreateDatabaseOperation(databaseRecord, 1)
    //endregion
    
    //region Step_2
    err := globalDocumentStore.Maintenance().Server().Send(createDatabaseOperation)
    if err != nil {
        if _, ok := err.(*ravendb.ConcurrencyError); ok {
            fmt.Printf("Database '%s' already exists\n", databaseRecord.DatabaseName)
            return nil
        }
        fmt.Printf("createDatabaseOperation failed with %s\n", err)
        return err
    }
    //endregion

    return nil
}
//endregion

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/advanced/createDatabase/CreateDatabase.java
package net.ravendb.demo.advanced.createDatabase;

//region Usings
import net.ravendb.client.exceptions.ConcurrencyException;
import net.ravendb.client.serverwide.DatabaseRecord;
import net.ravendb.client.serverwide.operations.CreateDatabaseOperation;
//endregion
import net.ravendb.demo.common.DocumentStoreHolder;

public class CreateDatabase {

    public void run() {
        String databaseName = "someDb";

        //region Demo
        try {
            //region Step_1
            DatabaseRecord databaseRecord = new DatabaseRecord(databaseName);
            CreateDatabaseOperation createDatabaseOperation = new CreateDatabaseOperation(databaseRecord);
            //endregion

            //region Step_2
            DocumentStoreHolder.store.maintenance().server().send(createDatabaseOperation);
            //endregion
        } catch (ConcurrencyException e) {
            // Database already exists
        }
        //endregion
    }

}
